components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -129,11 +141,19 @@ class BrowserIcons @Suppress("LongParameterList") constructor(
         // (1) First prepare the request.
         val request = prepare(context, preparers, initialRequest)
 
-        // (2) Then try to load an icon.
-        val (icon, resource) = load(context, request, loaders, decoders, desiredSize)
+        // (2) Check whether icons should be downloaded in background.
+        val updatedLoaders = loaders.map {
+            when (it is HttpIconLoader) {
+                true -> if (initialRequest.waitOnLoad) it else backgroundHttpIconLoader
+                false -> it
+            }
```

Comment:
nit: For a boolean an if statement may suffice? :)

`if (it is HttpIconLoader && initialRequest.waitOnLoad)`